modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/form_instance_records_search_container.jsp

```diff
@@ -124,22 +124,32 @@ PortletURL portletURL = ddmFormViewFormInstanceRecordsDisplayContext.getPortletU
 
 						</c:when>
 						<c:when test="<%= StringUtil.equals(ddmFormField.getType(), DDMFormFieldTypeConstants.CHECKBOX) %>">
-							<c:choose>
-								<c:when test='<%= StringUtil.equals(ddmFormViewFormInstanceRecordsDisplayContext.getColumnValue(ddmFormFieldsMap.get(ddmFormField.getName()), StringPool.BLANK, ddmFormFieldValuesMap.get(ddmFormField.getName())), LanguageUtil.get(ddmFormViewFormInstanceRecordsDisplayContext.getDefaultLocale(formInstanceRecord), "true")) %>'>
-									<liferay-ui:search-container-column-icon
-										cssClass="ddm-icon-check table-cell-expand"
-										icon="check"
-										name="<%= ddmFormViewFormInstanceRecordsDisplayContext.getColumnName(ddmFormField) %>"
-									/>
-								</c:when>
-								<c:otherwise>
-									<liferay-ui:search-container-column-icon
-										cssClass="table-cell-expand"
-										icon="hr"
-										name="<%= ddmFormViewFormInstanceRecordsDisplayContext.getColumnName(ddmFormField) %>"
-									/>
-								</c:otherwise>
-							</c:choose>
+
+							<%
+							String values = ddmFormViewFormInstanceRecordsDisplayContext.getColumnValue(ddmFormFieldsMap.get(ddmFormField.getName()), StringPool.BLANK, ddmFormFieldValuesMap.get(ddmFormField.getName()));
+
+							String lowerCaseValues = StringUtil.toLowerCase(values);
+
+							String[] listValues = lowerCaseValues.split(", ");
+
+							StringBuilder translatedValues = new StringBuilder();
+
+							for (String value : listValues) {
+								String translatedValue = LanguageUtil.get((Locale)request.getLocale(), value);
+
+								translatedValues.append(translatedValue);
+
+								translatedValues.append(", ");
+							}
+
+							translatedValues.setLength(translatedValues.length() - 2);
+							%>
```

Comment:
@danvb112, can we move this logic to a dedicated method in the class `ddmFormViewFormInstanceRecordsDisplayContext`?

Moving it to the back-end it's easy to maintain and also allows the possibility to add unit tests, which isn't possible with the logic in JSP. :wink: 

Moreover, Liferay Portal API already provides some util methods that can assist you, please see the code below to get some inspiration:
```
public String getTranslatedColumnValues(String columnValues) {
	String[] translatedColumnValues = Arrays.stream(columnValues.split(","))
		.map(columnValue -> columnValue.trim())
		.map(columnValue -> columnValue.toLowerCase())
		.map(columnValue -> LanguageUtil.get((Locale)request.getLocale(), columnValue))
		.toArray(String[]::new);

	return StringUtil.merge(translatedColumnValues, StringPool.COMMA);
}
```

Reply:
Thanks for this tip @natocesarrego :smile: